addons/sale/models/sale_order.py

```diff
@@ -507,6 +511,36 @@ def _compute_amounts(self):
             order.amount_tax = tax_totals['tax_amount_currency']
             order.amount_total = tax_totals['total_amount_currency']
 
+    def _pay_term_is_early_discount_mixed(self):
+        self.ensure_one()
+        pay_term = self.payment_term_id
+        return pay_term and pay_term.early_discount and pay_term.early_pay_discount_computation == 'mixed'
+
+    def _add_lines_for_early_payment_discount_tax_correction(self):
+        self.ensure_one()
+        epd_lines = []
```

Comment:
```suggestion
            self.ensure_one()
            epd_lines = []
            if not (
                self.payment_term_id.early_discount
                and self.payment_term_id.early_pay_discount_computation == 'mixed'
            ):
                return epd_lines
```

Removes the need of `_pay_term_is_early_discount_mixed` and simplifies the logic.